src/dune_pkg/package_universe.mli

```diff
@@ -3,12 +3,26 @@ open! Import
 (** All of the packages in a dune project including local packages and packages
     in a lockdir. The lockdir is guaranteed to contain a valid dependency
     solution for the local packages. *)
-type t = private
-  { local_packages : Local_package.t Package_name.Map.t
-  ; lock_dir : Lock_dir.t
-  }
+type t
 
 val create
   :  Local_package.t Package_name.Map.t
   -> Lock_dir.t
   -> (t, User_message.t) result
+
+(** Raises a [User_error.E] if the lockdir doesn't contain a solution to the
+    local packages *)
+val create_exn : Local_package.t Package_name.Map.t -> Lock_dir.t -> t
```

Comment:
I see this function is only used once. So I would just inline it since it's just as well implemented on top of the existing `create`.

Reply:
done